src/main/resources/react4xp/table/Table.tsx

```diff
@@ -185,167 +107,127 @@ function Table(props: TableProps) {
     }
   }
 
-  function createTable() {
-    const { tableClass } = props.table
-
-    return (
-      <table className={tableClass} ref={tableRef}>
-        {addCaption()}
-        {table.thead!.map((t, index) => {
-          return (
-            <React.Fragment key={index}>
-              {addThead(index)}
-              {addTbody(index)}
-            </React.Fragment>
-          )
-        })}
-        {addTFoot()}
-      </table>
-    )
-  }
-
   function addCaption() {
-    const { caption } = table
+    const { caption } = currentTable
     if (caption) {
-      const hasNoteRefs = typeof caption === 'object'
-      return (
-        <caption data-noterefs={hasNoteRefs ? caption.noterefs : null} ref={captionRef}>
-          <div className='caption-text-wrapper'>
-            {hasNoteRefs ? caption.content : caption}
-            {hasNoteRefs ? addNoteRefs(caption.noterefs) : null}
-          </div>
-        </caption>
-      )
+      return typeof caption === 'object' ? caption.content : caption
```

Comment:
I think we need to keep the addNoteRefs(caption.noterefs) check here for caption text prop:
`{hasNoteRefs ? addNoteRefs(caption.noterefs) : null}`

Reply:
I see the `dataNoteRefs` code being passed in `createTable()`, which is the attribute (not the same as caption text prop). Good! 👏